pkg/actors/placement/placement.go

```diff
@@ -180,6 +180,7 @@ func (p *actorPlacement) Start(ctx context.Context) error {
 		defer p.shutdownConnLoop.Done()
 		ch := p.appHealthFn(ctx)
 		if ch == nil {
+			p.appHealthy.Store(true)
```

Comment:
This seems a bit off. I don't think we can do this because if the app has a HTTP app channel, `ch` will be non-nil.

Can you check `pkg/actors/actors.go`? Specifically, where `a.checker` is initialized:

https://github.com/JoshVanL/dapr/blob/a447ea75cd47e4c28256d6a5e4374dfadeb156ea/pkg/actors/actors.go#L278

Perhaps the way to fix this issue is to simply not initialize it if `len(hat) == 0`?

https://github.com/JoshVanL/dapr/blob/a447ea75cd47e4c28256d6a5e4374dfadeb156ea/pkg/actors/actors.go#L261

```go
	if len(hat) > 0 {
		a.checker, err = a.getAppHealthChecker()
		if err != nil {
			return fmt.Errorf("actors: couldn't create health check: %w", err)
		}
	}
```

Reply:
We should be covering this scenario in the `noentities` test.

https://github.com/dapr/dapr/pull/7394/files/a447ea75cd47e4c28256d6a5e4374dfadeb156ea#diff-104d2d64415550335dafcda3cb0730e5d866648f66f25ce74a40faffad722b58

Reply:
Hold on, then if `ch` is already `nil` when entities is empty.... There's nothing we need to do in this PR? Was the issue already fixed? (And just needed tests?)

Reply:
When we start placement, we already set appHealthy to true:

https://github.dev/JoshVanL/dapr/blob/a447ea75cd47e4c28256d6a5e4374dfadeb156ea/pkg/actors/placement/placement.go#L159

And if `ch` is nil, then this can never be set to false

(And if that weren't the case, workflow tests would have failed already sinced they don't implement /healthz)

Reply:
Without the code change made in this PR, the new test `noentities` will fail. i.e. actors will never become ready.

Reply:
It works on my machine without this line. Please try it!

Reply:
@JoshVanL Please, double check this because we already have E2E to validate that sidecars that don't serve actors types can still invoke other actors.

Reply:
BTW, happy to take the integration test part of it.

Reply:
Done! I have just included the integration tests now.